### PR TITLE
fixed some small bugs and added new flare ability.

### PR DESCRIPTION
**Pull request by Castelias**

### CODE REVIEW

File: Assets/Prefab/Player/Player.prefab [hunk 1]
```diff
@@ -238,7 +238,7 @@ GameObject:
   m_Component:
   - component: {fileID: 8333540880121230563}
   m_Layer: 10
-  m_Name: ShootingPoint_Back
+  m_Name: ShootingPoint_Backasd
```
Comment (piskorzm) [inline Assets/Prefab/Player/Player.prefab:241]: wut?

File: Assets/Scripts/Kamikaze.cs [hunk 2]
```diff
@@ -36,6 +41,18 @@ private void FixedUpdate()
 
                 _rb.velocity = transform.up * speed;
             }
+            else if (player !=null && flare != null)
+            {
```
Comment (piskorzm) [inline Assets/Scripts/Kamikaze.cs:45]: This logic could be simplified to 
if (player != null) {
    if (flare == null) {
        .....
    } 
   else { 
        .....
   }
}

File: Assets/Scripts/Flare.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if(collision.gameObject.tag == "Kamikaze_Enemy")
+        {
+            Destroy(gameObject);
+           
+            
+        }
+    }
+
```
Comment (piskorzm) [inline Assets/Scripts/Flare.cs:28]: You don't need to put so much space between the lines.

File: Assets/Scripts/Kamikaze.cs (hunk 2 above)
Comment (Castelias) [reply Assets/Scripts/Kamikaze.cs:45]: wyjebane jajca

File: Assets/Scripts/PlayerHitScanShooting.cs [hunk 4]
```diff
@@ -19,29 +19,29 @@ public class PlayerHitScanShooting : MonoBehaviour
         private Rigidbody2D _rigidBody;
         private float _nextShotTime = -100f;
 
-        private void Awake()
+       /* private void Awake()
```
Comment (PeterPwlk) [inline Assets/Scripts/PlayerHitScanShooting.cs:22]: Remove commented code if not used

File: Assets/Scripts/PlayerBullet.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            if (!_renderer.isVisible)
+            {
+                Destroy(gameObject);
+            }
+            _rb.AddForce(transform.up*moveSpeed, ForceMode2D.Impulse);
+        }
+
+        private void OnTriggerEnter2D(Collider2D collision)
+        {
+            Enemy enemy = GetComponent<Enemy>();
```
Comment (piskorzm) [inline Assets/Scripts/PlayerBullet.cs:33]: IOIOIOIOIOIOIOIO GIHUBOWA POLICJAAAA

File: Assets/Scripts/PlayerHitScanShooting.cs [hunk 6]
```diff
@@ -14,32 +14,32 @@ public class PlayerHitScanShooting : MonoBehaviour
...
         [SerializeField] private float timeBetweenShots = .25f;
         
-        //Input
+        
         private PlayerInput _playerInput;
         private Rigidbody2D _rigidBody;
         private float _nextShotTime = -100f;
 
-       /* private void Awake()
+        private void Awake()
```
Comment (PeterPwlk) [inline Assets/Scripts/PlayerHitScanShooting.cs]: delete this

File: Assets/Scripts/PlayerHitScanShooting.cs [hunk 7]
```diff
@@ -14,32 +14,32 @@ public class PlayerHitScanShooting : MonoBehaviour
...
-        }*/
+        }
 
         private void Start()
         {
             _rigidBody = GetComponent<Rigidbody2D>();
         }
         
-      /*  private void OnEnable()
+        private void OnEnable()
```
Comment (PeterPwlk) [inline Assets/Scripts/PlayerHitScanShooting.cs]: and this

File: Assets/Scripts/PlayerHitScanShooting.cs [hunk 8]
```diff
@@ -14,32 +14,32 @@ public class PlayerHitScanShooting : MonoBehaviour
...
         {
             _playerInput.Enable();
         }
 
         private void OnDisable()
```
Comment (PeterPwlk) [inline Assets/Scripts/PlayerHitScanShooting.cs]: and this

**Pull request merged**